src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import duke.command.DukeCommandWithArgs;
+import duke.exception.DukeStorageException;
+import duke.exception.InvalidCommandException;
+import duke.task.TaskList;
+
+public class Duke {
+    private static final String SAVE_FILE_LOCATION = "duke-task-list.txt";
+    private final Ui ui;
+    private final TaskList taskList;
+    private final Storage storage;
+    private final Parser parser;
+
+    public static void main(String[] args) {
+        Duke duke = new Duke();
+        duke.start();
+    }
+
+    public Duke() {
+        this.ui = new Ui(System.in, System.out);
+        this.storage = new Storage(SAVE_FILE_LOCATION);
+        TaskList taskList;
+        try {
+            taskList = storage.loadTaskList();
+        } catch (DukeStorageException e) {
+            ui.outputLine(e.getMessage());
+            taskList = new TaskList();
+        }
+        this.taskList = taskList;
+        this.parser = new Parser();
+    }
+
+    /**
+     * Starts duke.
+     */
+    public void start() {
+        ui.printWelcomeMessage();
+        ui.printHelp();
+        while (ui.shouldContinue()) {
+            String commandStr = ui.nextCommand();
```

Comment:
I can understand this but I am not sure if we should use standard word String or call it `commandInput` the Str instead as it will be hard to pronounce this way. 
```suggestion
            String commandInput = ui.nextCommand();
```